app/views/notes/show.html.erb

```diff
@@ -86,6 +86,10 @@
     </table>
   <% end %>
 
+  <% if current_user && ( current_user.role == "admin" || current_user.role == "moderator") && !(@node.has_tag('locked')) %>
+    <%= link_to "Lock Note", "/notes/lock_note/#{@node.id}" %>
```

Comment:
So i think it would need to be a JavaScript call instead of a direct route, in this case; see how it's done in this menu:

![image](https://user-images.githubusercontent.com/24359/120357138-1f109300-c2d3-11eb-89d6-eef97c056a67.png)

The functionality is buried quite deep here, but i think fundamentally it uses a JS function called `add_tag()`

Here's the menu itself:

https://github.com/publiclab/plots2/blob/main/app/views/tag/_advanced_tagging.html.erb

And aha - here it is in JavaScript:

https://github.com/publiclab/plots2/blob/e646cfd248e46fe9cf11a2eb7860bbf29f949b7d/app/assets/javascripts/tagging.js#L108

This is great to see, @Manasa2850 - thank you! What do you think about simplifying and taking this approach?